### PR TITLE
control_request

### PR DESCRIPTION
**Pull request by aswath1**
# PR Details
## Description

To support this the j2735_convertor node needs to be updated for the new ROS messages and ASN1 spec.
## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [x ] My change requires a change to the documentation.
- [x ] I have updated the documentation accordingly.
- [x ] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: carma-messenger-core/j2735_convertor/src/control_request_convertor.cpp [hunk 1]
```diff
@@ -54,101 +90,187 @@ void convert(const j2735_msgs::ControlBounds& in_msg, cav_msgs::ControlBounds& o
...
     }
   }
 
   return true;
 }
 
-void convert(const cav_msgs::ControlBounds& in_msg, j2735_msgs::ControlBounds& out_msg, const int8_t scale) {
+void convert(const cav_msgs::TrafficControlBounds& in_msg, j2735_msgs::TrafficControlBounds& out_msg, const int8_t scale) 
+{
   out_msg.oldest = (in_msg.oldest.sec * units::MS_PER_S) + (in_msg.oldest.nsec / units::NS_PER_MS_INT);
```
Comment (MishkaMN) [inline carma-messenger-core/j2735_convertor/src/control_request_convertor.cpp:122]: here also oldest should be minutes

File: carma-messenger-core/j2735_convertor/src/control_request_convertor.cpp [hunk 2]
```diff
@@ -54,101 +90,187 @@ void convert(const j2735_msgs::ControlBounds& in_msg, cav_msgs::ControlBounds& o
...
-  double scaled_z = in_msg.offsets[2] / pow(10, scale);
+  out_msg.reflat = in_msg.reflat * units::TENTH_MICRO_DEG_PER_DEG;
+  out_msg.reflon = in_msg.reflon * units::TENTH_MICRO_DEG_PER_DEG;
+
+  double scaled_deltax0 = in_msg.offsets[0].deltax / pow(10, scale);
+  double scaled_deltay0 = in_msg.offsets[1].deltay / pow(10, scale);
+  double scaled_deltax1 = in_msg.offsets[0].deltax / pow(10, scale);
+  double scaled_deltay1 = in_msg.offsets[1].deltay / pow(10, scale);
+  double scaled_deltax2 = in_msg.offsets[0].deltax / pow(10, scale);
+  double scaled_deltay2 = in_msg.offsets[1].deltay / pow(10, scale);
```
Comment (MishkaMN) [inline carma-messenger-core/j2735_convertor/src/control_request_convertor.cpp:131]: I am confused by the indices here. In my opinion, you would access them with the same order you are declaring your variable? e.g. in_msg.offsets[0].deltax ... in_msg.offsets[0].deltay... ain_msg.offsets[1].deltax ... in_msg.offsets[1.deltay etc

File: carma-messenger-core/j2735_convertor/src/control_request_convertor.cpp [hunk 3]
```diff
@@ -54,101 +90,187 @@ void convert(const j2735_msgs::ControlBounds& in_msg, cav_msgs::ControlBounds& o
...
 
-  out_msg.offsets[0] = scaled_x;
-  out_msg.offsets[1] = scaled_y;
-  out_msg.offsets[2] = scaled_z;
+  out_msg.offsets[0].deltax=scaled_deltax0;
+  out_msg.offsets[1].deltay=scaled_deltay0;
+  out_msg.offsets[0].deltax=scaled_deltax1;
+  out_msg.offsets[1].deltay=scaled_deltay1;
+  out_msg.offsets[0].deltax=scaled_deltax2;
+  out_msg.offsets[1].deltay=scaled_deltay2;
```
Comment (MishkaMN) [inline carma-messenger-core/j2735_convertor/src/control_request_convertor.cpp:151]: same indices confusion as described above

**Comment by aswath1** [comment]
https://github.com/usdot-fhwa-stol/carma-platform/issues/754

**Pull request merged**